src/core/mac/mac.hpp

```diff
@@ -917,6 +910,8 @@ class Mac : public InstanceLocator, private NonCopyable
     TimeMilli mCslTxFireTime;
 #endif
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+    uint8_t mIsCslEnabled : 1;
```

Comment:
It makes sense to me to move things like `IsCslSupported` from Mac to Mle. But I'm not in favor of adding a data copy in the Mac class. We can directly access the method in Mle class from the Mac class. I don't see any benefit of adding a data copy as a class member here.

If for code simplicity, we can use local variable in function body. Thoughts?

Reply:
I think the main reason to add these two variable here is preventing repeated invalid calls to the API `mLinks.SetCslParams()`. Only when the state of `mIsCslEnabled` or  `mIsCslCapable` is changed, the Mac layer will call the `mLinks.SetCslParams()` to enable or disable the CSL receiver. This is useful when we move the CSL receiver to the RCP later.

Reply:
Make sense to me.